### ISSUE TITLE
"Usage example" not working with iOS build

### ISSUE THREAD

**Issue by TheCertain** [SamVerschueren/gulp-cordova-icon#12]
After successfully incorporating most of the `gulp-cordova-*` plugins into my build process, `gulp-cordova-icon` is giving me a hard time :-(

So I finally tried out the simple usage example of the readme file (only substituting android with ios).

```
var gulp = require('gulp'),
    create = require('gulp-cordova-create'),
    icon = require('gulp-cordova-icon'),
    ios = require('gulp-cordova-build-ios');

gulp.task('build', function() {
    return gulp.src('dist')
    .pipe(create())
    .pipe(icon('res/my-icon.png'))
    .pipe(ios())
    .pipe(gulp.dest('build'));
});
```

The gulp task finishes without complaints. Also, the folder `.cordova/res` contains the icon. However, the app itself still features the standard cordova icon.

Maybe I am missing something here, but since I basically used the most basic example, I am not sure what to do. Any help would be greatly appreciated!

Two other (probably unrelated) things I noticed about the usage example:

```
android = require('gulp-cordova-android')
```

should probably read

```
android = require('gulp-cordova-build-android')
```

Also, I noticed that after performing the gulp task, I not only end up with a directory `./.cordova`, but also an empty directory `./build/.cordova`

**Comment by SamVerschueren** [comment]
> Also, I noticed that after performing the gulp task, I not only end up with a directory ./.cordova, but also an empty directory ./build/.cordova

With iOS, you don't have to pipe to `gulp.dest('build')`. The reason you have to do that for Android is because it will store the apk files in the `build` directory. With iOS, it's not yet possible to generate an IPA file directly (https://github.com/SamVerschueren/gulp-cordova-build-ios/issues/1).

So this would be the example gulp task

``` js
var gulp = require('gulp'),
    create = require('gulp-cordova-create'),
    icon = require('gulp-cordova-icon'),
    ios = require('gulp-cordova-build-ios');

gulp.task('build', function() {
    return gulp.src('dist')
    .pipe(create())
    .pipe(icon('res/my-icon.png'))
    .pipe(ios());
});
```

> Also, the folder .cordova/res contains the icon. However, the app itself still features the standard cordova icon.

Can you take a look at `.cordova/platforms/ios/<project>/Resources/icons`. What icon is generated over there? I tested it here and the icon I provide to the `icon` action is the icon being generated in all the different sizes.

**Comment by TheCertain** [comment]
Thanks for the quick reply!

I found the problem. First of all, I was completely ignorant about the `errorHandlingStrategy` parameter (sometimes it helps to read the manual all the way to the end ;-) ).

The problem is: it actually matters whether I have installed GraphicsMagick or ImageMagick. I only had ImageMagick installed, and `gm` does not seem to have an automatic "fallback" mechanism, but expects GraphicsMagick by default.

So maybe it would be a good idea to either implement such a fallback in `icon.js`, maybe like so:

```
var gm = require('gm');

gm(1, 1).write('test.png', function (err) {
    if(err) {
        gm = require('gm').subClass({imageMagick: true});
    } else {
        fs.unlink('test.png');
    }
});
```

The easier solution would be of course to just change the documentation to require the user to install GraphicsMagick... ;-)

Anyway, thanks for your work, you are making my app developing attempts a lot easier than I expected them to be!
